File: src/com/watnapp/etipitaka/plus/E_TipitakaApplication.java
package com.watnapp.etipitaka.plus;

import android.app.Application;
import android.content.Context;
import com.watnapp.etipitaka.plus.helper.BookDatabaseHelper;
import com.watnapp.etipitaka.plus.model.History;

/**
 * Created with IntelliJ IDEA.
 * User: sutee
 * Date: 27/5/2013
 * Time: 14:39

 */
public class E_TipitakaApplication extends Application {

  private BookDatabaseHelper.Language language = BookDatabaseHelper.Language.THAI;
  private History history;
  private static Context context;

  public BookDatabaseHelper.Language getLanguage() {
    return language;
  }

  public void setLanguage(BookDatabaseHelper.Language language) {
    this.language = language;
  }

  public History getHistory() {
    return history;
  }

  public void setHistory(History history) {
    this.history = history;
  }

  public void onCreate(){
    super.onCreate();
    E_TipitakaApplication.context = getApplicationContext();
  }

  public static Context getAppContext() {
    return ETipitakaApplication.context;
  }

}

File: app/src/main/java/com/watnapp/etipitaka/plus/ETipitakaApplication.kt
package com.watnapp.etipitaka.plus

import android.app.Application
import android.content.Context
import com.watnapp.etipitaka.plus.helper.BookDatabaseHelper
import com.watnapp.etipitaka.plus.model.History
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin

/**
 * Created with IntelliJ IDEA.
 * User: sutee
 * Date: 27/5/2013
 * Time: 14:39
 *
 */
open class ETipitakaApplication : Application() {
    var language = BookDatabaseHelper.Language.THAI
    var history: History? = null

    override fun onCreate() {
        super.onCreate()
        startKoin {
            androidContext(this@ETipitakaApplication)
            modules(appModule)
        }
    }
}